interfaces.go

```diff
@@ -5,14 +5,31 @@
 package optimize
 
 // A Method can optimize an objective function.
+//
+// It uses a reverse-communication interface design represented by the Init()
+// and Iterate() methods to pass and retrieve information from the client. Its
+// aim is to isolate the method from user-supplied information and data and
+// their representation, and from common operations like checking for (various
+// types of) convergence and maintaining statistics.
+//
+// When Method needs the client to perform an operation like function
```

Comment:
I would say something like "A Method can command an Evaluation or a MajorIteration operation. An evaluation operation is one or more of the Evaluaton operations (FunctionEvaluation, GradientEvaluation, etc.) which can be combined with the bitwise or operator. In an evaluation operation, the requested routines will be evaluated at the value specified in Location.X. The corresponding fields of Location will be filled with the results from the routine. Alternately, a method can declare a MajorIteration. In a MajorIteration, the values in Location are interpreted as a new minimum. Convergence of the optimization (GradientThreshold, etc.) will be checked using this new minimum."